### PR TITLE
make citation key optional

### PR DESCRIPTION
**Pull request by aluex**
Hey Alin, my Mendeley is also suffering from some issues. Then I recalled you mentioned this repo (a year ago maybe). Let's see if this one combined with a sync netdrive would make my life easier. ;)

### CODE REVIEW

File: citationkeys/misc.py [hunk 1]
```diff
@@ -98,7 +99,7 @@ def cks_to_tuples(ck_bib_dir, cks, verbosity):
             #print(bibtex.entries)
             #print("Comments: ")
             #print(bibtex.comments)
-            bib = bibtex.entries[0]
+            bib = defaultdict(lambda key: '', bibtex.entries[0])
```
Comment (alinush) [inline citationkeys/misc.py]: Hey Alex, pardon my Python ignorance, but what does this do that the previous line didn't?

File: citationkeys/urlhandlers.py [hunk 2]
```diff
@@ -83,7 +83,12 @@ def download_pdf_andor_bib(opener, user_agent, pdfurl, destpdffile, biburl, dest
 
 def dlacm_handler(opener, soup, parsed_url, ck_bib_dir, destpdffile, destbibfile, parser, user_agent, verbosity):
     path = parsed_url.path.split('/')[2:]
-    doi = path[0] + '/' + path[1]
+    if len(path) > 1:
+        doi = path[0] + '/' + path[1]
+    elif 'doid' in parsed_url.query:
+        doi = '10.1145/' + parsed_url.query.split('=')[1] # 10.1145/doid
```
Comment (alinush) [inline citationkeys/urlhandlers.py:88]: So I'm guessing this is a type of URL that arises for some papers that we couldn't handle before?
Do you have two examples of such URLs? Maybe we can add a comment here with those URLs so we can remember this later.

**Comment by aluex** [comment]
Also added arxiv support.

File: citationkeys/misc.py (hunk 1 above)
Comment (aluex) [reply citationkeys/misc.py]: defaultdict will return the default value when the key does not exist. Sometimes BibTeX returned by the websites have missing items.

File: citationkeys/urlhandlers.py (hunk 2 above)
Comment (aluex) [reply citationkeys/urlhandlers.py:88]: Let's mark this as TODO. I'm not sure about what 10.1145 means.

**Pull request merged**